components/purchase_modal/purchase_modal.tsx

```diff
@@ -224,6 +224,10 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
             if (products[key].billing_scheme === BillingSchemes.FLAT_FEE) {
                 flatFeeProducts.push(tempEl);
             } else {
+                // only show enterprise to customers upgrading from professional
+                if (currentProduct.sku === 'cloud-professional') {
+                    return;
+                }
```

Comment:
Why filter here for an specific product if it filters by product price below?

Reply:
ok